### ISSUE TITLE
FID measurements are invalid: two independent bugs in evaluate_fid

### ISSUE THREAD

**Issue by nikhilr612** [nikhilr612/tinyflow#10]
## Summary

Every FID number this repo has produced is invalid. There are **two independent bugs** in
`metrics.evaluate_fid`, both of which silently produce a plausible-looking float.

## Bug 1 — FID is computed on single-pixel-row images

`metrics.py:100`:

```python
saveat=diffrax.SaveAt(ts=jnp.array([1.0])),
...
return sol.ys[:, 0]
```

With `SaveAt(ts=...)`, `sol.ys` has shape `(len(ts),) + y0.shape` = `(1, 64, 64, 3)`. The
intended index is `sol.ys[0]`. `sol.ys[:, 0]` slices **axis 1** instead, giving `(1, 64, 3)` —
one row of pixels. Under `vmap` that becomes `(B, 1, 64, 3)`, and

```python
samples_299 = jax.image.resize(samples, (bs, 299, 299, 3), "bilinear")
```

does **not** error, because the rank still matches (4 == 4). It bilinearly stretches a 1x64
strip to 299x299.

So the recorded FIDs (275.6, 284.2, 258.9, 256.9, 283.5, 272.4, 331.7) were measured on
vertically-smeared single rows. `ImageFM.sample` indexes correctly (`trajectory[0, -1]`) — only
the hand-duplicated ODE term in `metrics.py` has this, which is the strongest argument for
de-duplicating it (see #12).

## Bug 2 — input range mismatch with `fidax`

`fidax.fid.FrechetInceptionDistance.update` documents `imgs: ... range [0, 1]` and internally
does:

```python
imgs = imgs.astype(self.model_dtype) * 2.0 - 1.0  # Scale to [-1, 1]
```

`metrics.py` passes images already in `[-1, 1]`, which fidax then maps to `[-3, 1]`. Inception
sees out-of-distribution inputs. Both real and fake go through the same transform so the
ranking is partly preserved, but the absolute number is meaningless and the features are
degraded.

## Additionally

- Generated samples are not clipped before the resize. Real features come from bounded data;
  fake ones come from an unbounded model (no final activation, by design), so the metric partly
  measures overshoot rather than image content.
- FID at `n_samples=5000` is biased. Two FIDs computed at different `n` are not comparable, so
  `n_samples` must be recorded alongside the metric.

## Proposed fix

Delete the hand-rolled ODE term and call a shared `model.generate(...)` (see #12),
which removes Bug 1 along with the code around it. Then:

```python
samples = jnp.clip(model.generate(noise, n_steps=n_steps), -1.0, 1.0)
samples_01 = (samples + 1.0) * 0.5          # fidax expects [0, 1]
samples_299 = jax.image.resize(samples_01, (bs, 299, 299, 3), "bilinear")
```

and apply the same `[0,1]` conversion to the real-statistics path in `compute_real_stats`.
**This invalidates the cached `.preprocessed/real_stats.npz`** — change the cache path so it
recomputes rather than relying on a manual delete.

**Comment by nikhilr612** [comment]
Fixed: the duplicated hand-rolled ODE integration in metrics.py was removed; FID sampling now goes through the shared sampler path (commit 7ad2eed / a074a97, training.py + models/imagefm.py). Confirmed no `sol.ys[:, 0]` mis-indexing remains.
